pandas/core/computation/ops.py

```diff
@@ -509,7 +509,9 @@ def _disallow_scalar_only_bool_ops(self) -> None:
 
 
 def isnumeric(dtype) -> bool:
```

Comment:
Can we replace this with `pandas.core.dtypes.common.is_numeric_dtype`.

Reply:
`pandas.core.dtypes.common.is_numeric_dtype` looks suitable. `isnumeric` function may be removed and `__init__` of `class Div(BinOp)` may be changed as follow:

```diff
-        if not isnumeric(lhs.return_type) or not isnumeric(rhs.return_type):
+        if not is_numeric_dtype(lhs.return_type) or not is_numeric_dtype(
+            rhs.return_type
+        ):
```
( `is_numeric_dytpe` added in imports of course)

@SiddheshBangar test may look like this:

``` python
def test_extension_array_eval():
    # GH#58748
    df = DataFrame({"a": pd.array([1, 2, 3]), "b": pd.array([4, 5, 6])})
    result = df.eval("a / b")
    expected = Series([0.25, 0.40, 0.50])
    tm.assert_series_equal(result, expected)
```